### PR TITLE
Update graph search header

### PR DESCRIPTION
**Pull request by fakob**
* updated the graph search header
  * now it shows the name of the selected graph
  * added save option
  * moved the share button inside the bar
  * added a create new/empty graph button
  * made it work for small screens
* added share option to context menu

Screenshot 2023-09-27 at 19 38 24
Screenshot 2023-09-27 at 19 38 14

### CODE REVIEW

File: src/components/Search.tsx [hunk 1]
```diff
@@ -21,48 +25,197 @@ import match from 'autosuggest-highlight/match';
...
+              props.setShowSharePlayground(true);
+            }}
+          >
+            Share
+          </Button>
+          {props.isLoggedIn && (
+            <Button
+              variant="text"
+              size="small"
+              title="Logour from Github"
```
Comment (magnificus) [inline src/components/Search.tsx:95]: small typo

File: src/PPStorage.tsx [hunk 2]
```diff
@@ -268,6 +268,17 @@ export default class PPStorage {
     }
   }
 
+  async getGraphNameFromDB(graphId = undefined): Promise<undefined | string> {
```
Comment (magnificus) [inline src/PPStorage.tsx:271]: This function has a lot of different behaviors, maybe you should be forced to send in graphId and return always string? (if there is no graph found then can return "GRAPH_NAME_NOT_FOUND or something"

**Review by magnificus — APPROVED** [review]
cool

File: src/components/Search.tsx (hunk 1 above)
Comment (fakob) [reply src/components/Search.tsx:95]: Good catch!

File: src/PPStorage.tsx (hunk 2 above)
Comment (fakob) [reply src/PPStorage.tsx:271]: Sounds good

**Pull request merged**